public/js/gogs.js

```diff
@@ -1035,4 +1037,18 @@ $(window).load(function () {
             }
         }).trigger('hashchange');
     }
+
+    // Repo clone url.
+    if ($('#repo-clone-url').length > 0) {
+        switch (localStorage.getItem('repo-clone-protocol')) {
+            case 'ssh':
+                if ($('#repo-clone-ssh').click().length === 0) {
```

Comment:
It's quite simple. jQuery uses callbacks for everything, and returns the original jQuery object. This way you can write chained statements like this:

```js
// Add event listeners .click and .dblclick to #button1
$('#button1').click(function() { console.log('button1 click') })
             .dblclick(function() { console.log('button1 double-click') });
```



`#button1` is the selector, and `$('#button1')` the jQuery object, which is basically a special list with DOM-Elements and jQuery functions.

Calls like `$('#button1')` are quite heavy (performance) because of reasons, and as jQuery returns what we need anyway, I can just do the `.length` check afterwards. Benefit: the selector evaluation is done only once, and we don't have to create a new variable. Yay!